commons/src/test/java/io/aiven/kafka/tiered/storage/commons/chunkindex/ChunkIndexBuilderCommonTest.java

```diff
@@ -0,0 +1,128 @@
+    @BeforeEach
+    protected abstract void init();
+
+    @Test
+    void addChunkToAlreadyFinishedIndex() {
+        normalInitedChunkIndexBuilder.addChunk(113);
+        normalInitedChunkIndexBuilder.addChunk(113);
+        normalInitedChunkIndexBuilder.finish(113);
+
+        assertThatThrownBy(() -> normalInitedChunkIndexBuilder.addChunk(113))
+            .isInstanceOf(IllegalStateException.class)
+            .hasMessage("Cannot add chunk to already finished index");
+    }
+
+    @Test
+    void addNonPositiveSizedChunk() {
+        assertThatThrownBy(() -> normalInitedChunkIndexBuilder.addChunk(-1))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Transformed chunk size must be non-negative, -1 given");
+    }
+
+    @Test
+    void addSupposedToBeFinalChunkAsNonFinal() {
+        normalInitedChunkIndexBuilder.addChunk(113);
+        normalInitedChunkIndexBuilder.addChunk(113);
+
+        assertThatThrownBy(() -> normalInitedChunkIndexBuilder.addChunk(113))
+            .isInstanceOf(IllegalStateException.class)
+            .hasMessage("This must be final chunk. Call `finish` instead.");
+    }
+
+    @Test
+    void finishedAlreadyFinishedIndex() {
+        normalInitedChunkIndexBuilder.addChunk(113);
+        normalInitedChunkIndexBuilder.addChunk(113);
+        normalInitedChunkIndexBuilder.finish(113);
+
+        assertThatThrownBy(() -> normalInitedChunkIndexBuilder.finish(113))
+            .isInstanceOf(IllegalStateException.class)
+            .hasMessage("Cannot finish already finished index");
+    }
+
+    @Test
+    void finishWithNegativeSizedChunk() {
+        normalInitedChunkIndexBuilder.addChunk(113);
+        normalInitedChunkIndexBuilder.addChunk(113);
+
+        assertThatThrownBy(() -> normalInitedChunkIndexBuilder.finish(-1))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Transformed chunk size must be non-negative, -1 given");
+    }
+
+    @Test
+    void finishWithNonFinalChunk() {
+        normalInitedChunkIndexBuilder.addChunk(113);
+
+        assertThatThrownBy(() -> normalInitedChunkIndexBuilder.finish(113))
+            .isInstanceOf(IllegalStateException.class)
+            .hasMessage("This cannot be final chunk: not enough chunks to cover original file. "
+                + "Call `addChunk` instead.");
+    }
+
+    @Test
+    void findForNegativeOffset() {
+        normalInitedChunkIndexBuilder.addChunk(113);
+        normalInitedChunkIndexBuilder.addChunk(113);
+        final ChunkIndex index = normalInitedChunkIndexBuilder.finish(113);
+
+        assertThatThrownBy(() -> index.findChunkForOriginalOffset(-1))
+            .isInstanceOf(IllegalArgumentException.class)
+            .hasMessage("Offset must be non-negative, -1 given");
+    }
+
+    @Test
+    void findForEmptyFile() {
+        final ChunkIndex index = emptyFileChunkIndexBuilder.finish(0);
+
+        assertThat(index.findChunkForOriginalOffset(0)).isEqualTo(null);
```

Comment:
Nit:
```suggestion
        assertThat(index.findChunkForOriginalOffset(0)).isNull();
```

And below also